packages/key-management/src/util/ownSignatureKeyPaths.ts

```diff
@@ -59,6 +59,32 @@ const getStakingKeyPaths = (
   return paths;
 };
 
+/**
+ * Predicate that evaluates whether the given input set contains inputs that do not belong to the wallet.
+ *
+ * @param inputs The set of inputs to inspect.
+ * @param knownAddresses The wallet known address.
+ * @param inputResolver the input resolver.
+ */
+export const hasForeignInputs = async (
+  inputs: Cardano.TxIn[],
+  knownAddresses: GroupedAddress[],
+  inputResolver: Cardano.util.InputResolver
+) =>
+  inputs.some(async (input) => {
+    const ownAddress = await inputResolver.resolveInputAddress(input);
+    if (!ownAddress) return null;
```

Comment:
The logic seems to be inverted here: `hasForeignInputs` should be `true` if we find some input that is **not own**
```suggestion
    if (!ownAddress) return true;
```